presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -149,6 +154,7 @@ public OrcRecordReader(
         this.rowGroupStatisticsValidation = writeValidation.map(validation -> validation.createWriteStatisticsBuilder(includedColumns));
         this.stripeStatisticsValidation = writeValidation.map(validation -> validation.createWriteStatisticsBuilder(includedColumns));
         this.fileStatisticsValidation = writeValidation.map(validation -> validation.createWriteStatisticsBuilder(includedColumns));
+        this.systemMemoryUsage = requireNonNull(systemMemoryUsage, "systemMemoryUsage is null").newAggregatedMemoryContext();
```

Comment:
Please move `requireNonNull(systemMemoryUsage, "systemMemoryUsage is null")` to the requireNonNull checks at the beginning of this method. (We don't usually chain requireNonNulls with method calls)